app/helpers/script_helper.rb

```diff
@@ -22,21 +22,32 @@ def javascript_packs_tag_once(*names, prepend: false)
 
   alias_method :enqueue_component_scripts, :javascript_packs_tag_once
 
-  def render_javascript_pack_once_tags
+  def render_javascript_pack_once_tags(*names)
+    javascript_packs_tag_once(*names) if names.present?
     return if !@scripts
 
     # RailsI18nWebpackPlugin will generate additional assets suffixed per locale, e.g. `.fr.js`.
     # See: app/javascript/packages/rails-i18n-webpack-plugin/extract-keys-webpack-plugin.js
     regexp_locale_suffix = %r{\.(#{I18n.available_locales.join('|')})\.js$}
 
     locale_sources, sources = @scripts.flat_map do |name|
-      current_webpacker_instance.manifest.lookup_pack_with_chunks!(name, type: :javascript)
-    end.uniq.partition { |source| regexp_locale_suffix.match?(source) }
+      manifest.dig('entrypoints', name, 'assets', 'js')
+    end.uniq.compact.partition { |source| regexp_locale_suffix.match?(source) }
 
     javascript_include_tag(
       *locale_sources.filter { |source| source.end_with? ".#{I18n.locale}.js" },
       *sources,
     )
   end
+
+  private
+
+  def manifest
+    @manifest ||= begin
+      JSON.parse(File.read(MANIFEST_PATH))
+    rescue JSON::ParserError, Errno::ENOENT
+      {}
+    end
+  end
```

Comment:
Not a blocker, and this [essentially matches what Webpacker did anyways](https://github.com/rails/webpacker/blob/6614a4fc80db202690b7f4ce6772b239c193ab63/lib/webpacker/manifest.rb#L84), but: Wondering if there might be a more efficient way to cache the manifest _between_ requests, e.g. in-memory cache like Redis, or as part of application config (like we do with `IdentityConfig`).

Reply:
Would caching it cause issues in dev/test?

Reply:
For posterity: We spoke a bit on Slack, and will continue this at #5805.